sdk/src/network-client.ts

```diff
@@ -1198,24 +1198,53 @@ class AleoNetworkClient {
    * const transaction = await networkClient.waitForTransactionConfirmation(transactionId);
    */
   async waitForTransactionConfirmation(
-      transactionId: string,
-      checkInterval: number = 2000,  // Poll every 2 seconds
-      timeout: number = 45000        // Timeout after 45 seconds
-  ): Promise<Transaction> {
+    transactionId: string,
+    checkInterval: number = 2000,
+    timeout: number = 45000
+  ): Promise<"accepted" | "rejected"> {
     const startTime = Date.now();
-
-    return new Promise<Transaction>((resolve, reject) => {
+  
+    return new Promise((resolve, reject) => {
       const interval = setInterval(async () => {
+        const elapsed = Date.now() - startTime;
+  
+        if (elapsed > timeout) {
+          clearInterval(interval);
+          return reject(new Error("Transaction confirmation timed out"));
+        }
+  
         try {
-          // Replace with actual Aleo transaction lookup API
-          const transaction = <Transaction>await this.getTransactionObject(transactionId);
-          resolve(transaction);
-          if (Date.now() - startTime > timeout) {
+          const res = await fetch(`https://api.explorer.provable.com/v1/mainnet/transaction/confirmed/${transactionId}`);
+        
+          if (!res.ok) {
+            const text = await res.text();
+          
+            // Break early if the response indicates an invalid format
+            if (res.status >= 400 && res.status < 500 && text.includes("Invalid URL")) {
+              clearInterval(interval);
+              return reject(new Error(`Malformed transaction ID: ${text}`));
+            }
+          
+            // Log and continue polling for 404s or 5XX errors in case a tx doesn't exist yet
+            console.warn("Non-OK response (retrying):", res.status, text);
+            return;
+          }
+          
+          const data = await res.json();
+        
+          if (data?.status === "accepted") {
+            clearInterval(interval);
+            return resolve("accepted");
+          }
+        
+          if (data?.status === "rejected") {
             clearInterval(interval);
-            reject(new Error("Transaction confirmation timed out"));
+            return resolve("rejected");
```

Comment:
Here it should have an informative error such as this I think.

```suggestion
            reject(return reject(new Error(`Transaction ${transactoinID} was rejected by the network`));
```